packages/workflow/src/Cron.ts

```diff
@@ -0,0 +1,68 @@
+interface BaseTriggerTime<T extends string> {
+	mode: T;
+}
+
+type CronExpression = string;
+interface CustomTrigger extends BaseTriggerTime<'custom'> {
+	cronExpression: CronExpression;
+}
+
+interface EveryXTrigger<U extends string> extends BaseTriggerTime<'everyX'> {
+	unit: U;
+	value: number;
+}
+
+type EveryMinuteTrigger = BaseTriggerTime<'everyMinute'>;
+type EveryXMinutesTrigger = EveryXTrigger<'minutes'>;
```

Comment:
Nit: I find all the `-Trigger` suffixes redundant since it is clear `BaseTriggerTime` is the base interface.